### DISCUSSION TITLE
Discuss the date picking pattern

### DISCUSSION THREAD

**Discussion by johanstromqvist** [Shopify/polaris-react#7853] [Pattern documentation]
Welcome to the date picking pattern discussion!

### Here are some things you can do here:

💡 **Share ideas** on how to make this pattern, or its documentation, better.

❓ **Ask a specific question** about this pattern. 

---
Tips before posting:
- Remember Polaris and our discussions are public. If you work for Shopify and have an externally sensitive topic, please reach out in #polaris in Slack
- Please [report bugs here](https://github.com/Shopify/polaris/issues/new?assignees=&labels=%F0%9F%90%9BBug&template=ISSUE.md)
- [Contributions](https://polaris.shopify.com/contributing) are always welcome
- Learn more about [Polaris discussions](https://github.com/Shopify/polaris/discussions/6750)

💜👩‍🚀✌️ the Polaris team

**Comment by themarkappleby** [comment]
This is great! However, I'm wondering if it could be pushed even further?

For example, the code related to implementing [the single date pattern](https://polaris.shopify.com/patterns/date-picking) is 86 lines long. The end result however, is _extremely_ similar to [the native HTML `<input type="date">` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date) which can be achieved in a single line and leaves little room for implementation error.

What if the Polaris date picking pattern was packaged into a single component similar to `<input type="date">`?

**Reply by SeanyB** [reply]
Thank you for the suggestion, @themarkappleby. We've been thinking along the same lines. Some patterns make sense to have example code for folks to copy and modify, but others benefit from being wrapped as a component. We are starting to investigate how best to provide these alongside the core Polaris React library.

**Comment by benhickson** [comment]
Is there a typescript equivalent?

**Reply by sophschneider** [reply]
Hi @benhickson! Do you mean a typescript equivalent of the code example?

**Reply by benhickson** [reply]
yep!

**Reply by Alagaesia93** [reply]
I could use that as well, the provided one is full of errors

**Comment by abdulrafaykhatri** [comment]
I was wondering, what would be the best way to allow merchants to input a specific time along with the date? Some use cases, like scheduling orders, or setting promotions, might require an exact timestamp rather than just a date.

Would it make sense to bundle a time picker with this component and require time input when needed? That way, merchants wouldn’t have to rely on a separate field for time selection. Curious to hear thoughts on this!
